bindings/pyroot/pythonizations/test/sofie_gnn.py

```diff
@@ -0,0 +1,320 @@
+        decoder.Generate()
+        decoder.OutputGenerated()
+
+        output_transform = ROOT.TMVA.Experimental.SOFIE.RModel_GraphIndependent.ParseFromMemory(ep_model._output_transform._network, OutputGraphData, filename = "output_transform")
+        output_transform.Generate()
+        output_transform.OutputGenerated()
+
+        # Including the sofie generated models
+        ROOT.gInterpreter.Declare('#include "encoder.hxx"')
+        ROOT.gInterpreter.Declare('#include "core.hxx"')
+        ROOT.gInterpreter.Declare('#include "decoder.hxx"')
+        ROOT.gInterpreter.Declare('#include "output_transform.hxx"')
+
+        encoder_session = ROOT.TMVA_SOFIE_encoder.Session()
+        core_session = ROOT.TMVA_SOFIE_core.Session()
+        decoder_session = ROOT.TMVA_SOFIE_decoder.Session()
+        output_transform_session = ROOT.TMVA_SOFIE_output_transform.Session()
+
+        # Preparing the input data for running inference on sofie
+        input_data = ROOT.TMVA.Experimental.SOFIE.GNN_Data()
+        input_data.node_data = ROOT.TMVA.Experimental.AsRTensor(InputGraphData['nodes'])
+        input_data.edge_data = ROOT.TMVA.Experimental.AsRTensor(InputGraphData['edges'])
+        input_data.global_data = ROOT.TMVA.Experimental.AsRTensor(InputGraphData['globals'])
+
+        output_gn = ep_model(input_graphs, 2)
+
+        # running inference on sofie
+        data = CopyData(input_data)
+
+        encoder_session.infer(data)
+        latent0 = CopyData(data)
+        latent = data
+        output_ops = []
+        for _ in range(2):
+            core_input = ROOT.TMVA.Experimental.SOFIE.Concatenate(latent0, latent, axis=1)
+            core_session.infer(core_input)
+            latent = CopyData(core_input)
+            decoder_session.infer(core_input)
+            output_transform_session.infer(core_input)
+            output = CopyData(core_input)
+            output_ops.append(output)
+
+        for i in range(0, len(output_ops)):
+          output_node_data = output_gn[i].nodes.numpy()
+          output_edge_data = output_gn[i].edges.numpy()
+          output_global_data = output_gn[i].globals.numpy().flatten()
+
+          assert_almost_equal(output_node_data, np.asarray(output_ops[i].node_data))
+
+          assert_almost_equal(output_edge_data, np.asarray(output_ops[i].edge_data))
+
+          assert_almost_equal(output_global_data, np.asarray(output_ops[i].global_data))
+
+
+    def tearDown(self) :
+        filesToRemove = ['core','encoder','decoder','output_transform','gnn_network','graph_independent_network']
+        fileExist = True
+        for fname in filesToRemove:
+          if (not os.path.isfile(fname + '.hxx')) :
+              fileExist = False
+        #remove files when all files exist
+        if (fileExist):
+            for fname in filesToRemove:
+                os.remove(fname + '.hxx')
+                os.remove(fname + '.dat')
```

Comment:
This is not what I suggested. This function will be executed at the end of every test, not only once at the end of all tests. See the Python standard docs https://docs.python.org/3/library/unittest.html#unittest.TestCase.tearDownClass

Please change again the test

Reply:
Yes correct, the function will be executed after each test,  but the file removal is done only at the end, when all header files have been generated.